merlin-server/src/main/java/gov/nasa/jpl/aerie/merlin/server/services/GetSimulationResultsAction.java

```diff
@@ -97,9 +99,11 @@ public Map<String, List<Violation>> getViolations(final PlanId planId, final Sim
       throw new RuntimeException("Assumption falsified -- mission model for existing plan does not exist");
     }
 
-
     final var activities = new ArrayList<ActivityInstance>();
-    for (final var entry : results.simulatedActivities.entrySet()) {
+    final var simulatedActivities = this.simulationService
+      .get(planId, revisionData, r -> r.simulatedActivities)
```

Comment:
I'm noticing three calls to `simulationService.get` - any thoughts on the tradeoffs between keeping them separate, vs merging them into one call (and then pulling `simulatedActivities, `discreteProfiles`, and `realProfiles` out afterwards).

Reply:
Making one call also goes well with my note on #191 -- you can just pass `$ -> $` (or its better named equivalent, `Function.identity()`) to get an `Optional<SimulationResults>`.

Reply:
I do like the idea of condensing to one function call. My first attempt at this looks like
```java
final var results = this.simulationService
    .get(planId, revisionData, Function.identity());
```
followed by
```java
final var simulatedActivities = results
    .map(r -> r.simulatedActivities)
    .orElseGet(Collections::emptyMap);
```
for each field.

Is there a simpler way of extracting a field from an `Optional` that I'm overlooking?

Reply:
Sorry for the churn @skovati, sloppy thinking on my part when I added `SimulationService.get()`. If you want you can change it to:
```java
Optional<SimulationResults> get(final PlanId planId, final RevisionData revisionData);
```
(Thanks @Twisol)
Or use the `Function.identity()` approach you have above.

Then in `getViolations()` I would just call that once, with something like: `final var results$ = simulationService.get(planId, revisionData);`.

Reply:
A-ok with either option, changing `SimulationService.get()` would require a slight change to your work for [1832](https://github.com/NASA-AMMOS/aerie/pull/191) @pcrosemurgy. Happy to make that change, just let me know your preference.

Reply:
No strong opinion :) Whatever you see fit is good with me

Reply:
I'll stick with the generic `get` function in that case, barring additional input, since otherwise we'd have two methods in `SimulationService` that both return slightly different forms of sim results.

Reply:
Well, they both accomplish the same thing so you'd only want 1 either way :)
```java
<T> Optional<T> get(final PlanId planId, final RevisionData revisionData, final Function<SimulationResults, T> getter);
```
or
```java
Optional<SimulationResults> get(final PlanId planId, final RevisionData revisionData);
```

So using the first form you can do:
```java
final var simResults$ = simulationService.get(planId, revisionData, Function.identity());
final var simulatedActs = simResults$.map(r -> r.simulatedActivities).orElseGet(Map::of);
```

Using the second form you can return the same data with a slightly simpler call:
```java
final var simResults$ = simulationService.get(planId, revisionData);
final var simulatedActs = simResults$.map(r -> r.simulatedActivities).orElseGet(Map::of);
...
```

**TL;DR** either is really fine, sorry for causing confusion here with my weird `get()` interface.

Reply:
er yeah, I was referencing the already existing `getSimulationResults` in `SimulationService`, so changing `get()` to only return `SimulationResults` felt a little redundant (of course it isn't, since the whole point is we need this optional logic for uncached sims) while the current generic implementation offers slightly more functionality

Reply:
It sounds like you'll have two methods that return slightly different forms of sim results no matter what your choice here is, no? If they serve different use-cases, we ought to evolve to allow them to serve those use-cases effectively. If we eventually find we can unify them, that's some thing we can do later. If we eventually find they're impossible to unify, we accept that we have two ways of getting sim results and that there's a fundamental difference between them.

Whether you apply your getter to the return value yourself, or let the method call your getter for you, does not seem to be materially relevant to the fact that there are two ways to get results. My vote would be to remove the getter callback, because it doesn't serve the use-case any better than using Optional's own map method does.

Reply:
More meaningfully, the getter argument entices you into calling get() multiple times with different getters, doing extra work to get the same initial simulation results every time. From a "pit of success" perspective, removing the fetter argument avoids this enticement and makes the right thing the default.

Reply:
>It sounds like you'll have two methods that return slightly different forms of sim results no matter what your choice here is, no? If they serve different use-cases, we ought to allow them to evolve to serve those use-cases effectively. 

Sorry if I'm misunderstanding what you're saying but it sounds like you're saying we have the need for both forms for `get()`? 

Looking at where `SimulationService.get()` is used we only need one form (the `Optional<SimulationResults>` form being the most ergonomic as you point out).

Reply:
No, I was responding to @skovati's note that get() and getSimulationResults() get simulation results. They serve different use cases, so avoiding similarity for similarity's sake prevents them from evolving effectively to meet their respective use cases.

Reply:
Change has been made.

Reply:
Whoops, pushed changes before I saw your input @Twisol, happy to make that change

Reply:
Alright nongeneric version has been pushed!

Reply:
Looks great! I think we've addressed Matt's initial concern here, too.